Modules/database_manager.py

```diff
@@ -0,0 +1,232 @@
+                raise ValueError("")
+
+            condition_string: str = ""
+            for k, v in condition.items():
+                condition_string += f" {k}='{v}' {connector}"
+            if len(condition_string) > 0:
+                condition_string = condition_string[:-len(connector)]
+
+            value_string: str = ""
+            for k, v in values.items():
+                value_string += f" {k}='{v}',"
+
+            value_string = value_string[:-1]
+
+            self.connection.execute(f"UPDATE '{table_name}' SET {value_string} WHERE {condition_string}")
+
+        def _delete_row(self, table_name: str, connector: str, condition=None):
+            """
+            deletes the rows from the specified table and filters them by the specified conditions.
+                Supports 'AND' and 'OR'
+
+            Args:
+                table_name (str): name of table to delete from
+                connector (str): either 'AND' or 'OR'
+                condition (dict): a dict of conditions to filter the result.
+                    these will be connected via 'connector'  and in the end
+                    must be true for the given row to be deleted
+
+            Returns: None
+
+            Raises: ValueError
+                1. should connector not be 'OR' or 'AND'
+                2. When the given table does not exist
+            """
+            if condition is None:
+                condition = {}
+            if not self._has_table(table_name):
+                raise ValueError(f"Table {table_name} does not exist")
+
+            if not len(condition) > 0:
+                raise ValueError("No conditions were given!")
+
+            connector = connector.upper()
+            if connector not in ["AND", "OR"]:
+                raise ValueError(f"Connector {connector} not supported")
+
+            condition_string: str = ""
+            for k, v in condition.items():
+                condition_string += f"{k}='{v}' {connector} "
+            condition_string = condition_string[:-(len(connector) + 1)]
+
+            self.connection.execute(f"DELETE FROM '{table_name}' WHERE {condition_string}")
+
+    def get_fact(self, name: str, preferred_lang: str):
+        """
+        Return the fact for the given fact name and language.
+        If the preferred_language doesnt exist, it returns the fact for the english language.
+        If neither exist, it returns an empty string
+        Args:
+            name(str): name of the fact
+            preferred_lang (str): preferred language
+
+        Returns(str): fact for the given language, the english language or an empty string, if neither exist.
+
+        """
+        result = self.instance._select_rows("mecha3-facts", "AND", {"lang": preferred_lang, "fact":name})
+        if len(result) <= 0:
+            result = self.instance._select_rows("mecha3-facts", "AND", {"lang": "en", "fact":name})
+        return result[0][2] if len(result) > 0 else ""

```

Comment:
Am I right in assuming that this returns an empty string when no fact could be found?
Because I'm not sure that's so great. Raising an exception might be better so that the caller can wrap the whole shebang in a try except block without worrying too much about return value.
Though returning None could also be alright. Just something more concrete than an empty string.

Reply:
You are not only right in assuming it does, it actually does in a documented fashion 😛

Reply:
That's good, I hadn't noticed that on my small phone screen.
However it only answers the first part of my comment. I don't believe returning an empty string is very concise, though this is probably a matter of opinion.